### ISSUE TITLE
Basic Auth

### ISSUE THREAD

**Issue by jri-sp** [rawmind0/alpine-traefik#29]
Hi,

Is it possible to add this feature to your rancher image to have it in the catalog ?

https://github.com/rawmind0/rancher-traefik/issues/31

Do you prefer an issue in https://github.com/rancher/community-catalog ?

Best Regards

**Comment by rawmind0** [comment]
Hi @jri-sp,

yes, it's possible but i have to find time....Or i would be happy to accept a PR for that... :P

**Comment by jri-sp** [comment]
Hi @rawmind0 ,
Since my message I've seen that I've to wait the traefik release to change the number version in your Dockerfile. 
So I'll do a PR when the release come! that's the lesser of it.

Thanks for your answer!

**Comment by jri-sp** [comment]
Hi @rawmind0 ,

I've updated my traefik stack to "rawmind/alpine-traefik:1.3.3" in Rancher  but my label `traefik.frontend.auth.basic` don't work... :( 
Do you know if it's implemented in your image ?
Have you tested it ?

Regards

**Comment by rawmind0** [comment]
Hi @jri-sp ...

With this image, you could use 2 different rancher integration. Traefik built-in rancher integration, ``` TRAEFIK_RANCHER_ENABLE="false" ``` and the unofficial rancher integration with rancher-traefik. In rancher-traefik is not yet implemented the ``` traefik.frontend.auth.basic ``` label, but yes in traefik built-in rancher integration.

Traefik built-in rancher integration has a "bug" that needs an admin api key for work. The bug is addressed to Traefik team and it will be addressed in next traefik release. Once it would be solved, i'll change traefik catalog package in rancher, just to use the built-in integration and not rancher-traefik. :)

Best regards...

**Comment by jri-sp** [comment]
Hi @rawmind0 

Ah ok! I'm confused, I didn't know the existence of this switch...
So I think I'll try to use the built-in integration when the bug will be solved.

Can you point me the issue number in Traefik Github if there is one ? If there is one I can follow it and don't disturb you again...

Best Regards,

**Comment by rawmind0** [comment]
Hi @jri-sp 

use the built-in integration make all the sense. This integration was done when built-in didn't exist. ;)

You could use the built-in integration right now, it just works, but you need an admin user and traefik could see all the environments inside your rancher...not so good....

The problem is at https://github.com/containous/traefik/blob/v1.3.3/provider/rancher/rancher.go with the function, listRancherEnvironments... It should be like that...
```
func listRancherEnvironments(client *rancher.RancherClient) []*rancher.Environment {
    var projectList = []*rancher.Project{}
    projects, err := client.Project.List(withoutPagination)
    if err != nil {
        log.Errorf("Cannot get Provider Projects %+v", err)
    }
    for k := range projects.Data {
        projectList = append(projectList, &projects.Data[k])
    }
    return projectList
}
```
The function at traefik v1.33 is different, but it's already changed at traefik master branch. This change will be included in the next traefik release.

Best regards....

**Comment by jri-sp** [comment]
Hi @rawmind0, is the function listRancherEnvironments is ok now ? (with 1.3.6)

The function is not exactly the same that you've quoted but I think it's good, isn't it?
Can I use built-in integration safely now ?

Regards

**Comment by rawmind0** [comment]
Hi @jri-sp ....
the function is changed with this PR, https://github.com/containous/traefik/pull/1563/files , merged in traefik v1.4.0-rc1.

Updated alpine-traefik image to v1.4.0-rc1... :)

**Comment by jri-sp** [comment]
Ah thank you very much! I'll try it in my dev environment!

**Comment by edbizarro** [comment]
I've tried to upgrade my stack with this v1.4.0-rc1 tag and add the label but no success

![screenshot from 2017-09-01 18-54-39](https://user-images.githubusercontent.com/84926/29989059-1a2ffdee-8f47-11e7-8a9f-73c392cb0c4b.png)

**Comment by gad0lin** [comment]
Hi,

what should proper label if I want to pass it?
Can I pass basic auth setting per service, in case of that example   http://rancher.com/traefik-active-load-balancer-on-rancher/ inside web-test compose files? 
Can i use some other authentication mechanism via that model like oauth2?

best regards...

**Comment by rawmind0** [comment]
Merged PR #42.. Added statistics and basic auth... Thanks @Zevran... :)

Please, reopen issue if you have any trouble....
